src/inmanta/ast/variables.py

```diff
@@ -178,15 +171,7 @@ def variable_resume(
     ) -> None:
         if self.resultcollector:
             variable.listener(self.resultcollector, self.owner.location)
-
-        if variable.is_ready():
-            self.write_target(variable)
-        else:
-            # reschedule on the variable's completeness
-            RawUnit(queue_scheduler, resolver, {self: variable}, self, override_exception_location=False)
```

Comment:
I quite like that we now no longer need to reschedule for plain variable dereferencing. Only `is defined` still needs it because it needs to actually process the value rather than just copy it.